### PR TITLE
fix: improve code robustness

### PR DESCRIPTION
**Pull request by lekhovitsky**
* make `calcDebtAndCollateral` total debt calculation consistent with full repayment
* handle quoted tokens more reasonably in `manageDebt`
* remove the nasty 1 in quota keeper
* remove annoying invariant comments

### CODE REVIEW

File: contracts/libraries/QuotasLogic.sol [hunk 1]
```diff
@@ -32,7 +32,7 @@ library QuotasLogic {
         returns (uint128)
     {
         // `quoted` is `uint96`, and `cumulativeIndex / RAY` won't reach `2 ** 32` in reasonable time, so casting is safe
-        return quoted <= 1 ? 0 : uint128(uint256(quoted) * (cumulativeIndexNow - cumulativeIndexLU) / RAY); // U:[QL-2]
+        return quoted == 0 ? 0 : uint128(uint256(quoted) * (cumulativeIndexNow - cumulativeIndexLU) / RAY); // U:[QL-2]
```
Comment (0xmikko) [inline contracts/libraries/QuotasLogic.sol]: Why not:

 return uint128(uint256(quoted) * (cumulativeIndexNow - cumulativeIndexLU) / RAY); // U:[QL-2]

**Review by 0xmikko — APPROVED** [review]

File: contracts/libraries/QuotasLogic.sol (hunk 1 above)
Comment (lekhovitsky) [reply contracts/libraries/QuotasLogic.sol]: indeed

File: contracts/credit/CreditManagerV3.sol [hunk 2]
```diff
@@ -716,15 +712,15 @@ contract CreditManagerV3 is ICreditManagerV3, SanityCheckTrait, ReentrancyGuardT
...
 
-        cdd.accruedInterest = cdd.cumulativeQuotaInterest;
-        cdd.accruedInterest += CreditLogic.calcAccruedInterest({
+        cdd.accruedInterest = CreditLogic.calcAccruedInterest({
             amount: cdd.debt,
             cumulativeIndexLastUpdate: cdd.cumulativeIndexLastUpdate,
             cumulativeIndexNow: cdd.cumulativeIndexNow
-        }); // U:[CM-21]
+        });
+        cdd.accruedFees = currentCreditAccountInfo.quotaFees + cdd.accruedInterest * feeInterest / PERCENTAGE_FACTOR;
```
Comment (3sGgpQ8H) [inline contracts/credit/CreditManagerV3.sol:720]: Consider using the `mulDiv` function or preventing phantom overflow in some other way.

File: contracts/credit/CreditManagerV3.sol [hunk 3]
```diff
@@ -716,15 +712,15 @@ contract CreditManagerV3 is ICreditManagerV3, SanityCheckTrait, ReentrancyGuardT
...
 
-        cdd.accruedFees = currentCreditAccountInfo.quotaFees;
-        cdd.accruedFees += cdd.accruedInterest * feeInterest / PERCENTAGE_FACTOR; // U:[CM-21]
+        cdd.accruedInterest += cdd.cumulativeQuotaInterest; // U:[CM-21]
+        cdd.accruedFees += cdd.cumulativeQuotaInterest * feeInterest / PERCENTAGE_FACTOR; // U:[CM-21]
```
Comment (3sGgpQ8H) [inline contracts/credit/CreditManagerV3.sol:723]: And here too.

File: contracts/libraries/QuotasLogic.sol [hunk 4]
```diff
@@ -32,7 +32,7 @@ library QuotasLogic {
         returns (uint128)
     {
         // `quoted` is `uint96`, and `cumulativeIndex / RAY` won't reach `2 ** 32` in reasonable time, so casting is safe
-        return quoted <= 1 ? 0 : uint128(uint256(quoted) * (cumulativeIndexNow - cumulativeIndexLU) / RAY); // U:[QL-2]
+        return uint128(uint256(quoted) * (cumulativeIndexNow - cumulativeIndexLU) / RAY); // U:[QL-2]
```
Comment (3sGgpQ8H) [inline contracts/libraries/QuotasLogic.sol:35]: Consider using the mulDiv function or preventing phantom overflow in some other way.

**Pull request merged**